index.js

```diff
@@ -42,10 +42,15 @@ function rateLimitPlugin (fastify, settings, next) {
     whitelist: globalParams.whitelist
   }
 
-  if (settings.redis) {
-    pluginComponent.store = new RedisStore(settings.redis, 'fastify-rate-limit-', globalParams.timeWindow)
+  if (settings.store) {
+    const Store = settings.store
+    pluginComponent.store = new Store(globalParams.timeWindow, 'fastify-rate-limit-')
```

Comment:
Why this interface as a constructor?
the `fastify-rate-limit-` was an utill for redis.

What do you think about `globalParams` ?
So user can do a lot of (strange 😄) things as an increment doubled for users that are not in the whitelist

Reply:
@Eomm Agree, that is a better call. I adjusted the PR accordingly and made the examples more explicit.